qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,408 @@
+        freq_bounds: Optional[Tuple[float, float]] = None,
+        offset_bounds: Tuple[float, float] = (-2, 2),
+        plot: bool = True,
+        ax: Optional["AxesSubplot"] = None,
+        **kwargs,
+    ) -> Tuple[AnalysisResult, None]:
+        """Analyze the given data by fitting it to a Gaussian.
+
+        Args:
+            experiment_data: The experiment data to analyze.
+            data_processor: The data processor with which to process the data. If no data
+                processor is given a singular value decomposition of the IQ data will be
+                used for Kerneled data and a conversion from counts to probabilities will
+                be done if Discriminated data was measured.
+            meas_level: The measurement level of the experiment data.
+            meas_return: Whether single-shot (the default) or average data is returned by the
+                experiment.
+            amp_guess: The amplitude of the Gaussian function, i.e. :math:`a`. If not
+                provided, this will default to -1 or 1 depending on the measured values.
+            sigma_guesses: The guesses for the standard deviation of the Gaussian distribution.
+                If it is not given this will default to an array of ten
+                points linearly spaced between zero and width of the x-data.
+            freq_guess: A guess for the frequency of the peak :math:`x0`. If not provided
+                this guess will default to the location of the highest or lowest point of
+                the y-data depending on the y-data.
+            offset_guess: A guess for the magnitude :math:`b` offset of the fit function.
+                If not provided, the initial guess defaults to the median of the y-data.
+            amp_bounds: Bounds on the amplitude of the Gaussian function as a tuple of
+                two floats. The default bounds are (-1, 1).
+            sigma_bounds: Bounds on the standard deviation of the Gaussian function as a tuple
+                of two floats. The default values are (0, frequency range).
+            freq_bounds: Bounds on the center frequency as a tuple of two floats. The default
+                values are (min(frequencies) - df, max(frequencies) - df).
+            offset_bounds: Bounds on the offset of the Gaussian function as a tuple of two floats.
+                The default values are (-2, 2).
+            plot: If True generate a plot of fitted data.
+            ax: Optional, matplotlib axis to add the plot to.
+            kwargs: Trailing unused function parameters.
+
+        Returns:
+            The analysis result with the estimated peak frequency and the plots if a plot was
+            generated.
+
+        Raises:
+            QiskitError:
+                - If the measurement level is not supported.
+                - If the fit fails.
+        """
+
+        # Pick a data processor.
+        if data_processor is None:
+            if meas_level == MeasLevel.CLASSIFIED:
```

Comment:
I don't think this is necessary defined in the analysis. This branch is routine of data processing thus this should be integrated into the processor. However, in terms of simplification, I prefer removing this branching, and just set default processor and meas conditions. If user changed meas conditions, they should provide appropriate processor.

Reply:
Indeed. I created a classmethod in the data processor. See here: 717dce0. We can extend this method in the future as we go along.